src/containers/SignUpScreen/index.js

```diff
@@ -2,36 +2,45 @@ import React, { memo, useCallback } from 'react';
 import { Text, View, Button } from 'react-native';
 import { useDispatch } from 'react-redux';
 import { object } from 'prop-types';
+import { isEmpty } from 'lodash';
 
 import SignUpForm from 'components/user/SignUpForm';
 import { signUp } from 'actions/userActions';
+import useSessionChangeEffect from 'hooks/useSessionChangeEffect';
 import translate from 'utils/i18n';
+import { MAIN_SCREEN } from 'constants/screens';
 import styles from './styles';
 
-const SignUpScreen = ({ navigation }) => {
+const SignUpScreen = memo(({ navigation }) => {
   const dispatch = useDispatch();
 
   const signUpRequest = useCallback(user => dispatch(signUp(user)), [dispatch]);
 
+  useSessionChangeEffect(
+    user => {
+      const userLoggedIn = !isEmpty(user);
+      if (userLoggedIn) {
+        navigation.navigate(MAIN_SCREEN);
+      }
+    },
+    [navigation]
+  );
```

Comment:
You could abstract these 3 hook implementations (Login and MainScreen) into their own hook.

Reply:
Yep, I like the idea